actors/builtin/market/deal.go

```diff
@@ -20,33 +29,55 @@ var PieceCIDPrefix = market0.PieceCIDPrefix
 // minimal deals that last for a long time.
 // Note: ClientCollateralPerEpoch may not be needed and removed pending future confirmation.
 // There will be a Minimum value for both client and provider deal collateral.
-//type DealProposal struct {
-//	PieceCID     cid.Cid `checked:"true"` // Checked in validateDeal, CommP
-//	PieceSize    abi.PaddedPieceSize
-//	VerifiedDeal bool
-//	Client       addr.Address
-//	Provider     addr.Address
-//
-//	// Label is an arbitrary client chosen label to apply to the deal
-//	// TODO: Limit the size of this: https://github.com/filecoin-project/specs-actors/issues/897
-//	Label string
-//
-//	// Nominal start epoch. Deal payment is linear between StartEpoch and EndEpoch,
-//	// with total amount StoragePricePerEpoch * (EndEpoch - StartEpoch).
-//	// Storage deal must appear in a sealed (proven) sector no later than StartEpoch,
-//	// otherwise it is invalid.
-//	StartEpoch           abi.ChainEpoch
-//	EndEpoch             abi.ChainEpoch
-//	StoragePricePerEpoch abi.TokenAmount
-//
-//	ProviderCollateral abi.TokenAmount
-//	ClientCollateral   abi.TokenAmount
-//}
-type DealProposal = market0.DealProposal
+type DealProposal struct {
+	PieceCID     cid.Cid `checked:"true"` // Checked in validateDeal, CommP
+	PieceSize    abi.PaddedPieceSize
+	VerifiedDeal bool
+	Client       addr.Address
+	Provider     addr.Address
+
+	// Label is an arbitrary client chosen label to apply to the deal
+	// TODO: Limit the size of this: https://github.com/filecoin-project/specs-actors/issues/897
```

Comment:
@laudiacay since we have a draft FIP started now is the perfect time to knock out this todo by enforcing a limit on this at the protocol level.  It could be added to the FIP which could increase in scope to "Deal Proposal Label improvements" or something like that.

To enforce a size all we need is validation code in the validate deal method that rejects proposals with too big labels with `ErrIllegalArgument`.  The tricky part is setting a good limit.  Investigating current labels in chain state will be informative (I can help with this).  Another helpful data point to look into is the cbor serialization limit on byte slices which I think is a few megabytes.

@whyrusleeping @ribasushi do you have insights into how the label is currently used / how you plan to use it?  Any recommendations on label size?

Reply:
errrr @ZenGround0 this is already implemented...?
https://github.com/filecoin-project/specs-actors/blob/v5.0.4/actors/builtin/market/policy.go#L26-L27

Reply:
okie dokie ill just take out the todo